--- a/security/selinux/avc.c
+++ b/security/selinux/avc.c
@@ -987,6 +987,11 @@
 				u8 driver, u8 xperm, unsigned flags,
 				struct av_decision *avd)
 {
+	if (flags & AVC_STRICT)
+		return -EACCES;
+
+	if (selinux_enforcing && !(avd->flags & AVD_FLAGS_PERMISSIVE))
+		return -EACCES;
 
 	avc_update_node(AVC_CALLBACK_GRANT, requested, driver, xperm, ssid,
 				tsid, tclass, avd->seqno, NULL, flags);
